File: chapter_two/selectionSort.js
const swap = (arr, indexA, indexB) => {
    let temp = arr[indexA]
    arr[indexA] = arr[indexB]
    arr[indexB] = temp
}
const selectionSort = arr => {
    for ( let i = 0; i < arr.length; i += 1 ) {
        let smallestIndex = i
        for ( let j = i + 1; j < arr.length; j += 1 ) {
            if ( arr[j] < arr[smallestIndex] ){
                smallestIndex = j
            }
        }
        swap(arr, smallestIndex, i)
    }
}

let testArr = []

for ( let i = 0; i < 100; i += 1 ) {
    let randomNum = Math.floor((Math.random() * 100))
    testArr.push(randomNum)
}

console.log(testArr)
selectionSort(testArr)
console.log(testArr)

File: chapter_two/selection_sort.py
import random

def selection_sort(arr):
    for i in range(len(arr)):
        smallest_index = i
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[smallest_index]:
                smallest_index = j
        arr[i], arr[smallest_index] = arr[smallest_index], arr[i]


test_arr = [random.randint(0,100) for i in range(100)]

print(test_arr)
selection_sort(test_arr)
print(test_arr)